### PR TITLE
Validate service account

### PR DESCRIPTION
**Pull request by tnsetting**
# Hey, I just made a Pull Request!

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [x] This PR has been incorporated in release note for the coming version
- [x] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java [hunk 1]
```diff
@@ -133,6 +139,13 @@ public static Builder newBuilder(DockerImageValidator dockerImageValidator) {
       }
     });
 
+    // Check if the service account contains space
+    cfg.serviceAccount().ifPresent(serviceAccount -> {
+      if (!validateServiceAccount(serviceAccount)) {
+        e.add("service account format is not valid");
```
Comment (narape) [inline styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java]: Let's make this message more informative, like:
```java
e.add("service account is not a valid email address: " + serviceAccount);
```

File: styx-common/src/test/java/com/spotify/styx/util/WorkflowValidatorTest.java [hunk 2]
```diff
@@ -247,6 +248,44 @@ public void shouldNotAllowComponentWithHash(String component) {
...
+            .serviceAccount(serviceAccount)
+            .build();
+
+    assertThat(sut.validateWorkflow(Workflow.create("test", configuration)),
+            contains("service account format is not valid"));
+  }
+
+  @Test
+  public void shouldRejectSpaceServiceAccountWithTailingSpace() {
+    String[] invalidServiceAccounts = {"sa@abc.com ", "sa@abc.com\n"};
```
Comment (narape) [inline styx-common/src/test/java/com/spotify/styx/util/WorkflowValidatorTest.java]: Please add a comment about the we needed to use a rudimentary for loop instead of a parametric tests because JUnitParams removes the trailing spaces

File: styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java [hunk 3]
```diff
@@ -133,6 +139,13 @@ public static Builder newBuilder(DockerImageValidator dockerImageValidator) {
       }
     });
 
+    // Check if the service account contains space
```
Comment (narape) [inline styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java]: Comment is not valid anymore. You can remove it

Comment (tnsetting) [reply styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java]: Removed.

File: styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java (hunk 1 above)
Comment (tnsetting) [reply styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java]: Addressed

File: styx-common/src/main/java/com/spotify/styx/testdata/TestData.java [hunk 4]
```diff
@@ -136,7 +136,7 @@
           .dockerImage("busybox")
           .dockerArgs(List.of("x", "y"))
           .secret(Secret.create("name", "/path"))
-          .serviceAccount("foo@bar.baz.quux")
+          .serviceAccount("foo@bar.baz.quu")
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/testdata/TestData.java]: Is this change needed?

Comment (tnsetting) [reply styx-common/src/main/java/com/spotify/styx/testdata/TestData.java]: Removed the change

File: styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java [hunk 5]
```diff
@@ -137,13 +144,19 @@ public static Builder newBuilder(DockerImageValidator dockerImageValidator) {
         upperLimit(e, timeout, maybeMaxRunningTimeout, "running timeout is too big");
       }
     });
-
+
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java]: Let's moves the empty spaces.

File: styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java [hunk 6]
```diff
@@ -161,6 +174,11 @@ public static Builder newBuilder(DockerImageValidator dockerImageValidator) {
     }
   }
 
+  private static boolean validateServiceAccount(String serviceAccount) {
+    Matcher matcher = VALID_EMAIL_ADDRESS_REGEX .matcher(serviceAccount);
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java]: `final` or just use `var`.
